pkg/component/networking/vpn/shoot/shoot.go

```diff
@@ -125,6 +126,127 @@ type vpnSecret struct {
 }
 
 func (v *vpnShoot) Deploy(ctx context.Context) error {
+	scrapeConfig := v.emptyScrapeConfig()
+	if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, v.client, scrapeConfig, func() error {
+		metav1.SetMetaDataLabel(&scrapeConfig.ObjectMeta, "prometheus", shoot.Label)
+		scrapeConfig.Spec = monitoringv1alpha1.ScrapeConfigSpec{
```

Comment:
Why do we not use `shoot.ClusterComponentScrapeConfigSpec(...)` here? Is this `ScrapeConfig` just too different?

Reply:
This is a config for `blackbox-exporter`, so somewhat "special". It indeed has some differences compared to the result of `shoot.ClusterComponentScrapeConfigSpec`, so I didn't use it.